workers/manifestsubjectbackfillworker.py

```diff
@@ -0,0 +1,115 @@
+from util.bytes import Bytes
+from util.log import logfile_path
+from util.migrate.allocator import yield_random_entries
+from workers.gunicorn_worker import GunicornWorker
+from workers.worker import Worker
+
+logger = logging.getLogger(__name__)
+
+WORKER_FREQUENCY = app.config.get("MANIFEST_SUBJECT_BACKFILL_WORKER_FREQUENCY", 60)
+
+
+class ManifestSubjectBackfillWorker(Worker):
+    """
+    Worker which backfills the newly added subject fields onto Manifest.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.add_operation(self._backfill_manifest_subject, WORKER_FREQUENCY)
+
+    def _backfill_manifest_subject(self):
+        try:
+            Manifest.select().where(Manifest.subject_backfilled == False).get()
+        except Manifest.DoesNotExist:
+            logger.debug("Manifest subject backfill worker has completed; skipping")
+            return False
+
+        iterator = yield_random_entries(
+            lambda: Manifest.select().where(Manifest.subject_backfilled == False),
+            Manifest.id,
+            250,
+            Manifest.select(fn.Max(Manifest.id)).scalar(),
+            1,
+        )
+
+        for manifest_row, abt, _ in iterator:
+            if manifest_row.subject_backfilled:
+                logger.debug("Another worker preempted this worker")
+                abt.set()
+                continue
+
+            logger.debug("Setting manifest subject for manifest %s", manifest_row.id)
+            manifest_subject = None
+            manifest_bytes = Bytes.for_string_or_unicode(manifest_row.manifest_bytes)
+
+            try:
+                parsed = parse_manifest_from_bytes(
+                    manifest_bytes, manifest_row.media_type.name, validate=False
+                )
+                subject = parsed.subject
+            except ManifestException as me:
+                logger.warning(
+                    "Got exception when trying to parse manifest %s: %s", manifest_row.id, me
+                )
+
+            updated = (
+                Manifest.update(
+                    subject=subject,
+                    subject_backfilled=True,
+                )
+                .where(Manifest.id == manifest_row.id, Manifest.subject_backfilled == False)
+                .execute()
+            )
+            if updated != 1:
+                logger.debug("Another worker preempted this worker")
+                abt.set()
+                continue
+
+        return True
+
+
+def create_gunicorn_worker():
+    """
+    follows the gunicorn application factory pattern, enabling
+    a quay worker to run as a gunicorn worker thread.
+
+    this is useful when utilizing gunicorn's hot reload in local dev.
+
+    utilizing this method will enforce a 1:1 quay worker to gunicorn worker ratio.
+    """
+    worker = GunicornWorker(
+        __name__, app, ManifestBackfillWorker(), features.MANIFEST_SIZE_BACKFILL
```

Comment:
- `ManifestBackfillWorker` should be `ManifestSubjectBackfillWorker`
- `MANIFEST_SIZE_BACKFILL` should be `MANIFEST_SUBJECT_BACKFILL`